### PR TITLE
bsdlib: Bump to version 0.6.1

### PR DESCRIPTION
**Pull request by evenl**
BSDLib bump to version 0.6.1

* Implement hostname verification.
* Report correct number of avaiable sockets for the application.
* Simplify documentation in bsd_limits.h, and remove unused macros.
* Fix bug that prevent the application to detect AGPS notifications.
* Enable TLS session ID caching.
* Fix several bugs in session cache socket option.
* Add nrf_setdnsaddr() to set the secondary DNS address.

Manifest pull request for NRF:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1821

### CODE REVIEW

File: bsdlib/doc/CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,17 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.6.1
+************
+
+* Implement hostname verification.
```
Comment (lemrey) [inline bsdlib/doc/CHANGELOG.rst]: `Implemented TLS hostname verification`.

File: bsdlib/doc/CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,17 @@ Changelog
...
+* Report correct number of avaiable sockets for the application.
+* Simplify documentation in bsd_limits.h, and remove unused macros.
+* Fix bug that prevent the application to detect AGPS notifications.
+* Enable TLS session ID caching.
+* Fix several bugs in session cache socket option.
```
Comment (lemrey) [inline bsdlib/doc/CHANGELOG.rst]: Merge both lines as:
`Implemented TLS session caching, enabled by default.`

File: bsdlib/doc/CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,17 @@ Changelog
...
+* Add nrf_setdnsaddr() to set the secondary DNS address.
```
Comment (lemrey) [inline bsdlib/doc/CHANGELOG.rst]: s/Add/Added

File: bsdlib/doc/CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,17 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.6.1
+************
+
+* Implement hostname verification.
+* Report correct number of avaiable sockets for the application.
+* Simplify documentation in bsd_limits.h, and remove unused macros.
```
Comment (lemrey) [inline bsdlib/doc/CHANGELOG.rst]: ~Simplify documentation in bsd_limits.h~ Removed unused `BSD_MAX_IP_SOCKET_COUNT` and `BSD_MAX_AT_SOCKET_COUNT` macros.

File: bsdlib/doc/CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,17 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.6.1
+************
+
+* Implement hostname verification.
+* Report correct number of avaiable sockets for the application.
```
Comment (lemrey) [inline bsdlib/doc/CHANGELOG.rst]: The number of available sockets hasn't changed so I suggest:
`Fixed a bug where the application couldn't allocate the 8th socket.`

**Comment by lemrey** [comment]
@ru-fu could you double check the changelog please?
@eibe85 could you please run the test?

**Comment by b-gent** [comment]
@lemrey , I implemented your comments, also grouped the points together (added, removed, fixed). Please check and squash.

File: bsdlib/doc/CHANGELOG.rst (hunk 1 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: ok

File: bsdlib/doc/CHANGELOG.rst (hunk 2 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: ok

File: bsdlib/doc/CHANGELOG.rst (hunk 3 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: Ok

File: bsdlib/doc/CHANGELOG.rst (hunk 4 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: ok

File: bsdlib/doc/CHANGELOG.rst (hunk 5 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: You fixed the issue by adding one socket so we can report 8 available sockets for the application.  So yes, the number of sockets has changed, but it's internal so will use your text.

**Pull request merged**